samples/snippets/sample_firewall.py

```diff
@@ -0,0 +1,159 @@
+
+    # Note that the default value of priority for the firewall API is 1000.
+    # If you check the value of `firewall_rule.priority` at this point it
+    # will be equal to 0. If you want to create a rule that has priority == 0,
+    # you'll need to explicitly set it so:
+
+    # firewall_rule.priority = 0
+
+    firewall_client = compute_v1.FirewallsClient()
+    op = firewall_client.insert(project=project_id, firewall_resource=firewall_rule)
+
+    op_client = compute_v1.GlobalOperationsClient()
+    op_client.wait(project=project_id, operation=op.name)
+
+    return
+# [END compute_firewall_create]
+
+
+# [START compute_firewall_patch]
+def patch_firewall_priority(project_id: str, firewall_rule_name: str, priority: int):
+    """
+    Modifies the priority of a given firewall rule.
+
+    Args:
+        project_id: project ID or project number of the Cloud project you want to use.
+        firewall_rule_name: name of the rule you want to modify.
+        priority: the new priority to be set for the rule.
+    """
+    firewall_rule = compute_v1.Firewall()
+    firewall_rule.priority = priority
+
+    # The patch operation doesn't require the full definition of a Firewall object. It will only update
+    # the values that were set in it, in this case it will only change the priority.
+    firewall_client = compute_v1.FirewallsClient()
+    operation = firewall_client.patch(project=project_id, firewall=firewall_rule_name, firewall_resource=firewall_rule)
+
+    operation_client = compute_v1.GlobalOperationsClient()
+    operation_client.wait(project=project_id, operation=operation.name)
+    return
+# [END compute_firewall_patch]
+
+
+# [START compute_firewall_delete]
+def delete_firewall_rule(project_id: str, firewall_rule_name: str):
+    """
+    Deleted a firewall rule from the project.
+
+    Args:
+        project_id: project ID or project number of the Cloud project you want to use.
+        firewall_rule_name: name of the firewall rule you want to delete.
+    """
+    firewall_client = compute_v1.FirewallsClient()
+    operation = firewall_client.delete(project=project_id, firewall=firewall_rule_name)
+
+    operation_client = compute_v1.GlobalOperationsClient()
+    operation_client.wait(project=project_id, operation=operation.name)
+    return
+# [END compute_firewall_delete]
+
+
+if __name__ == '__main__':
+    import google.auth
+    import google.auth.exceptions
+
+    try:
+        default_project_id = google.auth.default()[1]
+        print(f'Using project {default_project_id}.')
+    except google.auth.exceptions.DefaultCredentialsError:
+        print(
+            "Please use `gcloud auth application-default login` "
+            "or set GOOGLE_APPLICATION_CREDENTIALS to use this script."
+        )
+    else:
+        import uuid
+        rule_name = "firewall-sample-" + uuid.uuid4().hex[:10]
+        print(f'Creating firewall rule {rule_name}...')
+        create_firewall_rule(default_project_id, rule_name)
```

Comment:
For usage in default values presentation we should probably mention that this will get priority of 1000. This code is not part of the sample, so maybe it should be added in the create_firewall_rule - similar to how we did it in set_usage_reports examples.

Reply:
I added a comment, but I'm not sure how to gracefully add this to the create_firewall_rule sample, without adding the whole code block, which also contains call to functions from outside this sample.

I also expanded the comment in the `create_firewall_rule` function, to better explain what's going on around the priority.